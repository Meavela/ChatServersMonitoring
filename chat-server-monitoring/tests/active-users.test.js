import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import ActiveUsers from '../src/Components/active-users';

// Mock the component to simulate the API
jest.mock('../src/Components/active-users');

test('Display active users', () => {
    const {container} = render(<ActiveUsers />)
    expect(container.textContent).toMatch('http://127.0.0.1:7419/unittesthttp://127.0.0.1:6592/unitest2')
});

test('Not change active users', () => {
    const component = renderer.create(
        <ActiveUsers />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();  
});