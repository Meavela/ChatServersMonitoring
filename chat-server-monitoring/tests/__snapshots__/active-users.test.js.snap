// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not change active users 1`] = `
<div>
  <div
    className="MuiPaper-root MuiCard-root count-card MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root title"
    >
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          Active users
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h2"
      >
        2
      </h2>
      <div
        className="servers"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          On servers :
        </p>
        <ul
          className="MuiList-root MuiList-padding"
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root servers-items MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                http://127.0.0.1:7419/unittest
              </span>
            </div>
          </div>
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root servers-items MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                http://127.0.0.1:6592/unitest2
              </span>
            </div>
          </div>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
